src/Microsoft.ML.TorchSharp/NasBert/NerTrainer.cs

```diff
@@ -0,0 +1,363 @@
+
+namespace Microsoft.ML.TorchSharp.NasBert
+{
+    using TargetType = VBuffer<long>;
+
+    /// <summary>
+    /// The <see cref="IEstimator{TTransformer}"/> for training a Deep Neural Network(DNN) to classify text.
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [NER](xref:Microsoft.ML.TorchSharpCatalog.NameEntityRecognition(Microsoft.ML.MulticlassClassificationCatalog.MulticlassClassificationTrainers,System.String,System.String,System.String,Int32,Int32,Int32,Microsoft.ML.TorchSharp.NasBert.BertArchitecture,Microsoft.ML.IDataView)).
+    ///
+    /// ### Input and Output Columns
+    /// The input label column data must be a Vector of [string](xref:Microsoft.ML.Data.TextDataViewType) type and the sentence columns must be of type<xref:Microsoft.ML.Data.TextDataViewType>.
+    ///
+    /// This trainer outputs the following columns:
+    ///
+    /// | Output Column Name | Column Type | Description|
+    /// | -- | -- | -- |
+    /// | `PredictedLabel` | Vector of [key](xref:Microsoft.ML.Data.KeyDataViewType) type | The predicted label's index. If its value is i, the actual label would be the i-th category in the key-valued input label type. |
+    /// |  |  |
+    /// | -- | -- |
+    /// | Machine learning task | Multiclass classification |
+    /// | Is normalization required? | No |
+    /// | Is caching required? | No |
+    /// | Required NuGet in addition to Microsoft.ML | Microsoft.ML.TorchSharp and libtorch-cpu or libtorch-cuda-11.3 or any of the OS specific variants. |
+    /// | Exportable to ONNX | No |
+    ///
+    /// ### Training Algorithm Details
+    /// Trains a Deep Neural Network(DNN) by leveraging an existing pre-trained NAS-BERT roBERTa model for the purpose of name entity recognition.
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    ///
+    public class NerTrainer : NasBertTrainer<VBuffer<uint>, TargetType>
+    {
+        internal NerTrainer(IHostEnvironment env, NasBertOptions options) : base(env, options)
+        {
+        }
+
+        internal NerTrainer(IHostEnvironment env,
+            string labelColumnName = DefaultColumnNames.Label,
+            string predictionColumnName = DefaultColumnNames.PredictedLabel,
+            string sentence1ColumnName = "Sentence1",
+            int batchSize = 32,
+            int maxEpochs = 10,
+            IDataView validationSet = null,
+            BertArchitecture architecture = BertArchitecture.Roberta) :
+            this(env, new NasBertOptions
+            {
+                PredictionColumnName = predictionColumnName,
+                ScoreColumnName = default,
+                Sentence1ColumnName = sentence1ColumnName,
+                Sentence2ColumnName = default,
+                LabelColumnName = labelColumnName,
+                BatchSize = batchSize,
+                MaxEpoch = maxEpochs,
+                ValidationSet = validationSet,
```

Comment:
is this a validation dataset?  if yes, is this enables us to set validation set here instead of passing it to `Experiment.SetDataSet` ?

Reply:
This is a validation dataset, but its not plugged in internally anywhere yet. You could set it there and use it if you would like though. Will be hooked up in the next pr.